modules/NetworkSplitter3D/src/main/java/org/alexandrebarao/networksplitter3d/NetworkSplitter3D.java

```diff
@@ -0,0 +1,502 @@
+                    }
+                  
+                    
+                    // update new node coordinates 
+                    
+                    newY=newY + (zLevel*getZdistance()*getVerticalScale()); 
+                    
+                    n.setX((float)newX);
+                    n.setY((float)newY);
+                    n.setZ((float)newZ);
+             }
+             
+         }
+        
+        graph.readUnlockAll();
+     
+        endAlgo();
+    }
+
+    
+    
+   
+    @Override
+    public void endAlgo() {
+         executing = false;
+         Graph graph = graphModel.getGraphVisible();
+         
+         for (Node n : graph.getNodes()) {
+            n.setLayoutData(null);
+         }
+       
+        graph.readUnlockAll();
+        
+       
+    }
+
+    @Override
+    public boolean canAlgo() {
+        return executing;
+    }
+
+    @Override
+    public LayoutProperty[] getProperties() {
+        List<LayoutProperty> properties = new ArrayList<LayoutProperty>();
+        final String NETWORKSPLITTER3D = "Network Splitter 3D";
+        
+        
+        
+
+        try {
+            properties.add(LayoutProperty.createProperty(
+                    this, Integer.class,
+                    "Z-Maximum Level",
+                    NETWORKSPLITTER3D,
+                    "Z-maximum levels to compute node clusters. Node column '[z]' is needed in Gephi Data Laboratory. Note: you can use the RESET button to layout your original network.",
+                    "getZlevels", "setZlevels"));
+               
+             properties.add(LayoutProperty.createProperty(
+                    this, Integer.class,
+                    "Z-Distance Factor",
+                    NETWORKSPLITTER3D,
+                    "Layer distance between each Z-Level layout. Used to multiply Z-Scale. Note: you can use the RESET button to layout your original network.",
+                    "getZdistance", "setZdistance"));
+            
+             properties.add(LayoutProperty.createProperty(
+                    this, Float.class,
+                    "Z-Scale",
+                    NETWORKSPLITTER3D,
+                    "The vertical Z-Scale (pixels).  Note: you can use the RESET button to layout your original network.",
+                    "getVerticalScale", "setVerticalScale"));
+                     
+              properties.add(LayoutProperty.createProperty(
+                    this, Double.class,
+                    "Alfa",
+                    NETWORKSPLITTER3D,
+                    "The rotation angle (degrees from 0 to 360). Note: you can use the RESET button to layout your original network.",
+                    "getAlfa", "setAlfa"));
+              
+                         
+            
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return properties.toArray(new LayoutProperty[0]);
+    }
+
+     public double getDataLaboratoryValue(Node n, Column col) {

```

Comment:
Here you could check if the column is a subclass of ```Number``` Then use doubleValue() on the value.
Also, parsing Strings to Double might throw a parse exception. I suggest catching and ignoring that exception since that's not very likely to happen.